### PR TITLE
Dice modifiers and a limit on output of arrays.

### PR DESCRIPTION
**Pull request by drobati**
I still need to add modifiers to fudge dice.

Thoughts?

### CODE REVIEW

File: scripts/dice.coffee [hunk 1]
```diff
@@ -40,18 +46,30 @@ module.exports = (robot) ->
...
+          finalComma = if (results.length > 2) then "," else ""
+          last = results.pop()
+          "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        else
+          "I rolled a handful of dice, making #{total}."
+        if modifier
+          modified total, modifier
+
+modified = (total, modifier) ->
+  if modifier
```
Comment (skalnik) [inline scripts/dice.coffee]: I think you want a `?` after this to check for existence

File: scripts/dice.coffee [hunk 2]
```diff
@@ -40,18 +46,30 @@ module.exports = (robot) ->
...
-        finalComma = if (results.length > 2) then "," else ""
-        last = results.pop()
-        "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        if results.length < 10
+          finalComma = if (results.length > 2) then "," else ""
+          last = results.pop()
+          "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        else
+          "I rolled a handful of dice, making #{total}."
+        if modifier
```
Comment (skalnik) [inline scripts/dice.coffee]: Probably want a `?` after `modifier` here

File: scripts/dice.coffee [hunk 3]
```diff
@@ -40,18 +46,30 @@ module.exports = (robot) ->
...
         total = results.reduce (x, y) -> x + y
-        finalComma = if (results.length > 2) then "," else ""
-        last = results.pop()
-        "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        if results.length < 10
+          finalComma = if (results.length > 2) then "," else ""
+          last = results.pop()
+          "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        else
+          "I rolled a handful of dice, making #{total}."
```
Comment (skalnik) [inline scripts/dice.coffee]: This if/else pair probably needs to be set to some variable than you append the modifier stuff to afterwards and then return.

File: scripts/dice.coffee (hunk 1 above)
Comment (drobati) [inline scripts/dice.coffee]: So because it's a NaN the question mark didn't work. Fixed that I think.

File: scripts/dice.coffee (hunk 3 above)
Comment (drobati) [inline scripts/dice.coffee]: Ty.

**Comment by drobati** [comment]
@sshirokov Seems to work now. Does hubot have a unit test module?

**Comment by drobati** [comment]
Don't close this yet as I want to extend the modified function to the fudge dice rolls.

**Comment by drobati** [comment]
@sshirokov Okay I believe this has been resolved. I have tested manually with ''bin/hubot''.

**Comment by drobati** [comment]
@skalnik you can review it too

File: scripts/dice.coffee [hunk 4]
```diff
@@ -16,42 +16,68 @@
...
+#   drobati
+#   Added fudge dice and modifiers.
+#
+# Regex Tests:
+#   https://regex101.com/r/mC7vQ3/3
 
 module.exports = (robot) ->
   robot.respond /roll (die|one)/i, (msg) ->
     msg.reply report [rollOne(6)]
   robot.respond /roll dice/i, (msg) ->
```
Comment (skalnik) [inline scripts/dice.coffee]: A new line before all of these `robutt.respond` calls would definitely aid readability here :+1:

**Comment by sshirokov** [comment]
> Does hubot have a unit test module?

No, but I will give $5 to anyone that adds one.

**Comment by drobati** [comment]
I looked into the unit test stuff. It says there is a `make test` command but I couldn't figure out what the hell they were talking about.

I also saw people use `gulp` and some other library to do mocking. Just my two cents.

**Comment by skalnik** [comment]
If you want to have some sweet tests, that'd be rad. But the scripts are all basically untested

**Comment by sshirokov** [comment]
> If you want to have some sweet tests,

Yeah, I think tests should be a completely separate PR since we don't have any sort of setup for that at the moment. I wouldn't worry about it here.

**Comment by drobati** [comment]
Oh. I guess it was off-topic but I was totally thinking about a separate pull request. Also @skalnik my script is tested. I literally tested it manually 100 times today.

**Comment by sshirokov** [comment]
> I literally tested it manually 100 times today.

:star2:

**Pull request merged**